parsing_scripts/load_initial_data.py

```diff
@@ -0,0 +1,122 @@
+import os, json, sys
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+sys.path.append(BASE_DIR)
+os.environ['DJANGO_SETTINGS_MODULE'] = 'MacPorts.settings'
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "MacPorts.settings")
+
+import django
+
+django.setup()
+
+from ports.models import Port, Maintainer, Category
+from django.core.exceptions import ObjectDoesNotExist
+
+
+def open_portindex_json(path='portindex.json'):
+    with open(path, "r") as file:
+        ports = json.load(file)
+    return ports
+
+
+# Add All the Categories to the Database using bulk_create
+def load_categories_table(ports):
+    categories = set()
+    for port in ports:
+        try:
+            for category in port['categories']:
+                categories.add(Category(name=category))
+        except KeyError:
+            pass
+    batch = list(categories)
+    Category.objects.bulk_create(batch)
+    return
+
+
+def load_ports_and_maintainers_table(ports):
+    for port in ports:
+
+        # Add Ports to the Database One-by-One
+        new_port = Port()
+        try:
+            new_port.name = port['name']
+            new_port.portdir = port['portdir']
+            new_port.version = port['version']
+        except KeyError:
+            continue
+
+        new_port.variants = port.get('variants')
+        new_port.description = port.get('description')
+        new_port.homepage = port.get('homepage')
+        new_port.epoch = port.get('epoch')
+        new_port.platforms = port.get('platforms')
+        new_port.long_description = port.get('long_description')
+        new_port.revision = port.get('revision', 0)
+        new_port.closedmaintainer = port.get('closedmaintainer')
+        new_port.license = port.get('license')
+        new_port.installs_libs = port.get('installs_lib')
+        new_port.replaced_by = port.get('replaced_by')
+        new_port.save()
+
+        try:
+            new_port.categories.add(*port['categories'])
+        except KeyError:
+            pass
+
+        try:
+            for maintainer in port['maintainers']:
+                try:
+                    name = maintainer['email']['name'].lower()
+                    domain = maintainer['email']['domain'].lower()
+                    email_provided = True
+                except KeyError:
+                    name = None
+                    domain = None
+                    email_provided = False
```

Comment:
This looks ugly. Instead you should check whether the key exists. I assume the syntax is something like
```python
name = None
domain = None
github = None

if maintianer.has_key('email'):
    if maintainer['email'].has_key('name'):
        name = maintainer['email']['name'].lower()
    if maintainer['email'].has_key('domain'):
        domain = maintainer['email']['domain'].lower()
if maintainer.has_key('github'):
    github = maintainer['github'].lower()
```
That said, I'm not sure if I would downcase github handles. You can probably provide
```python
Maintainer.objects.get(..., github__iexact=github)
```
or something like that.

Reply:
> ```python
> name = None
> domain = None
> github = None
> 
> if maintianer.has_key('email'):
>     if maintainer['email'].has_key('name'):
>         name = maintainer['email']['name'].lower()
>     if maintainer['email'].has_key('domain'):
>         domain = maintainer['email']['domain'].lower()
> if maintainer.has_key('github'):
>     github = maintainer['github'].lower()
> ```

Thank You, using `get()` made it even simpler.